src/zarr/storage/_utils.py

```diff
@@ -74,11 +74,60 @@ def _join_paths(paths: Iterable[str]) -> str:
     """
     Filter out instances of '' and join the remaining strings with '/'.
 
-    Because the root node of a zarr hierarchy is represented by an empty string,
+    Parameters
+    ----------
+    paths : Iterable[str]
+
+    Returns
+    -------
+    str
+
+    Examples
+    --------
+    >>> _join_paths(["", "a", "b"])
+    'a/b'
+    >>> _join_paths(["a", "b", "c"])
+    'a/b/c'
     """
     return "/".join(filter(lambda v: v != "", paths))
 
 
+def _relativize_path(path: str, prefix: str) -> str:
+    """
+    Make a "\"-delimited path relative to some prefix. If the prefix is '', then the path is
+    returned as-is. Otherwise, the prefix is removed from the path as well as the separator
+    string "\".
+
+    If ``prefix`` is not the empty string and``path`` does not start with ``prefix``
+    followed by a "/" character, then an error is raised.
+
+    Parameters
+    ----------
+    path : str
+        The path to make relative to the prefix.
+    prefix : str
+        The prefix to make relative to.
+
+    Returns
+    -------
+    str
+
+    Examples
+    --------
+    >>> _relativize_paths("", "a/b")
+    'a/b'
+    >>> _relativize_paths("a/b", "a/b/c")
+    'c'
+    """
+    if prefix == "":
+        return path
+    else:
+        _prefix = prefix + "/"
```

Comment:
What behavior do we want for `prefix="<some-prefix>/"`, i.e. a prefix ending with a `/`?

IIUC, this will currently always raise a `ValueError`, unless `path` for some reason starts with `<some-prefix>//`

Maybe we should have `_prefix = prefix.rstrip("/") + "/"`? Or if you're confident the callers are careful about not providing a prefix with a trailing slash this is fine.

Reply:
I think I was assuming here that paths have already been normalized by [`normalize_path`](https://github.com/zarr-developers/zarr-python/blob/5ff3fbe5fe1488310301e9d2ae56a9880d1ddfb2/src/zarr/storage/_utils.py#L16), in which case prefixes cannot end with `"/"`. I should make this explicit in the docstring for the function here.